File: src/main/ModelsHandler.java
// ModelsHandler.java
import com.sun.net.httpserver.HttpExchange;
import com.sun.net.httpserver.HttpHandler;
import org.json.JSONObject;

import java.io.IOException;
import java.io.OutputStream;
import java.nio.charset.StandardCharsets;
import java.util.ArrayList;
import java.util.List;

public class ModelsHandler implements HttpHandler {
    public static final List<JSONObject> models = new ArrayList<>();

    static {
        models.add(new JSONObject().put("id", "gpt-4o").put("object", "model"));
        models.add(new JSONObject().put("id", "gpt-4o-mini").put("object", "model"));
        models.add(new JSONObject().put("id", "claude").put("object", "model"));
        models.add(new JSONObject().put("id", "claude-3-haiku").put("object", "model"));
        models.add(new JSONObject().put("id", "claude-3-5-claude").put("object", "model"));
        models.add(new JSONObject().put("id", "deepseek-r1").put("object", "model"));
    }
    @Override
    public void handle(HttpExchange exchange) throws IOException {
        // 仅允许GET请求
        if (!exchange.getRequestMethod().equalsIgnoreCase("GET")) {
            exchange.sendResponseHeaders(405, -1); // Method Not Allowed
            return;
        }

        try {

            // 构建响应JSON
            JSONObject responseJson = new JSONObject();
            responseJson.put("object", "list");
            responseJson.put("data", models);

            byte[] responseBytes = responseJson.toString().getBytes(StandardCharsets.UTF_8);
            exchange.getResponseHeaders().add("Content-Type", "application/json");
            exchange.sendResponseHeaders(200, responseBytes.length);

            OutputStream os = exchange.getResponseBody();
            os.write(responseBytes);
            os.close();
        } catch (Exception e) {
            e.printStackTrace();
            exchange.sendResponseHeaders(500, -1);
        }
    }
}

File: src/main/cpp/models_handler.cpp
#include "httplib.h"
#include "models_handler.h"
#include <iostream>

std::vector<nlohmann::json> ModelsHandler::models;

ModelsHandler::ModelsHandler() {
    initialize_models();
}

void ModelsHandler::initialize_models() {
    if (!models.empty()) return;
    models.push_back({{"id", "gpt-4o"}, {"object", "model"}});
    models.push_back({{"id", "gpt-4o-mini"}, {"object", "model"}});
    models.push_back({{"id", "claude"}, {"object", "model"}});
    models.push_back({{"id", "claude-3-haiku"}, {"object", "model"}});
    models.push_back({{"id", "claude-3-5-sonnet"}, {"object", "model"}});
    models.push_back({ {"id", "claude-3-7-sonnet"}, {"object", "model"} });
    models.push_back({ {"id", "sonar-reasoning-pro"}, {"object", "model"} });
    models.push_back({{"id", "deepseek-r1"}, {"object", "model"}});
}

void ModelsHandler::handle(const httplib::Request& req, httplib::Response& res) {
    if (req.method != "GET") {
        res.status = 405;
        return;
    }
    try {
        nlohmann::json response_json = {
            {"object", "list"},
            {"data", models}
        };
        res.set_content(response_json.dump(), "application/json");
        res.status = 200;
    } catch (const std::exception& e) {
        std::cerr << "Error in ModelsHandler: " << e.what() << std::endl;
        res.status = 500;
    }
}
